core/node/dns.go

```diff
@@ -16,18 +18,23 @@ var defaultResolvers = map[string]string{
 	"crypto.": "https://resolver.cloudflare-eth.com/dns-query",
 }
 
-func newResolver(url string) (madns.BasicResolver, error) {
+func newResolver(url string, opts ...doh.Option) (madns.BasicResolver, error) {
 	if !strings.HasPrefix(url, "https://") {
 		return nil, fmt.Errorf("invalid resolver url: %s", url)
 	}
 
-	return doh.NewResolver(url), nil
+	return doh.NewResolver(url, opts...), nil
 }
 
 func DNSResolver(cfg *config.Config) (*madns.Resolver, error) {
 	var opts []madns.Option
 	var err error
 
+	var dohOpts []doh.Option
+	if !cfg.DNS.MaxCacheTTL.IsDefault() {
+		dohOpts = append(dohOpts, doh.WithMaxCacheTTL(cfg.DNS.MaxCacheTTL.WithDefault(time.Duration(math.MaxUint32)*time.Second)))
```

Comment:
this will overflow...  do we actually need to set it?

Reply:
we need to set `WithDefault` to something. I considered using `0`, since there is an if before and we know the option is set.

`math.MaxUint32` is the default in [go-doh-resolver](https://github.com/libp2p/go-doh-resolver/blob/master/resolver.go#L63). We might want to update if there is an overflow.

Reply:
actually this is fine, duration is 64 bit and int32 max is not that big in that case.